### ISSUE TITLE
HLS can't correctly load `Haskell2010` pragma on ghc-9.2

### ISSUE THREAD

**Issue by July541** [haskell/haskell-language-server#2927]
GHC-9.2 implicitly enabled `GHC2021`. with a cabal project with `Haskell2010` as default language, it doesn't disable `GHC2021` as expected.

**Comment by googleson78** [comment]
Also encountering the same issue

**Comment by googleson78** [comment]
It reproduces with `stack` on my end, the OP seems to mention a cabal project though.

**Comment by fendor** [comment]
Reframing my question, what I meant is, does `stack build` have the same issue, or does it only happen in HLS? E.g., stack says it compiles as you expect, or does it also not load the pragma correctly?

**Comment by googleson78** [comment]
`stack` *does* respect the `language` field, at least with hpack, because things fail to compile with stack, but don't show errors when I run `haskell-language-server-wrapper` (e.g. `ImportQualifiedPost`)

**Comment by July541** [comment]
> I assume this is not reproducible with cabal?

Yes, all these happened in HLS, I tried to fix this, but the compile module in HLS is hard for me, I still don't know what happened while compiling.

**Comment by michaelpj** [comment]
Is this still happening?
